sql: CockroachDB panics when executing the ill-formed CREATE TABLE statement

**Describe the problem**

The latest version of the CockroachDB (v24.2.4 and the latest master commit: f15ee646b5111) crashes when executing the following query:

```sql
CREATE TABLE v00 (c01 INT);
CREATE TABLE v02 AS (SELECT * FROM v00 WINDOW window_name AS (ROWS c01 BETWEEN nextval ('abc', 'abc', 'abc') AND c01 PRECEDING));
```

**To Reproduce**

1. In operating system Ubuntu 20.04 LTS, download the pre-build CockroachDB binaries (v24.2.4) from [link](https://www.cockroachlabs.com/docs/releases/#v24-2)
2. Run ./cockroach demo, and then paste the PoC query to the cockroach cli environment.
3. Observe the crash and log the stack information.

**Expected behavior**
The CockroachDB should return error. 

**Additional data / screenshots**

This is the stack frame from the crash (v24.2.4):

```
*
* ERROR: a SQL panic has occurred while executing the following statement:
* CREATE TABLE v02 AS (SELECT * FROM movr.public.v00 WINDOW window_name AS (ROWS c01 BETWEEN nextval('abc', 'abc', 'abc') AND c01 PRECEDING))
*
*
* ERROR: a panic has occurred!
* could not find matching function overload for given arguments
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:961
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:127
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetSequenceFromFunc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:106
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:162
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RangeCond).Walk
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:501
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:840
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2138
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:157
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:456
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrameBound
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:250
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrame
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:221
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowDef
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:288
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1844
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ParenSelect).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1588
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleStmtVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2151
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:496
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:438
*   | github.com/cockroachdb/cockroach/pkg/sql.getFinalSourceQuery
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1183
*   | github.com/cockroachdb/cockroach/pkg/sql.newTableDescIfAs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1273
* Wraps: (2) while executing: CREATE TABLE _ AS (SELECT * FROM _ WINDOW _ AS (ROWS _ BETWEEN _('_', '_', '_') AND _ PRECEDING))
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:961
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:127
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetSequenceFromFunc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:106
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:162
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RangeCond).Walk
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:501
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:840
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2138
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:157
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:456
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrameBound
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:250
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrame
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:221
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowDef
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:288
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1844
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ParenSelect).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1588
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleStmtVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2151
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:496
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:438
*   | github.com/cockroachdb/cockroach/pkg/sql.getFinalSourceQuery
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1183
*   | github.com/cockroachdb/cockroach/pkg/sql.newTableDescIfAs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1273
* Wraps: (4) candidate pg code: 42804
* Wraps: (5) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetSequenceFromFunc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:106
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:162
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RangeCond).Walk
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:501
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:840
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2138
*   | github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences
*   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:157
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:456
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrameBound
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:250
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrame
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:221
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowDef
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:288
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1844
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ParenSelect).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1588
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleStmtVisit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2151
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:496
*   | github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:438
*   | github.com/cockroachdb/cockroach/pkg/sql.getFinalSourceQuery
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1183
*   | github.com/cockroachdb/cockroach/pkg/sql.newTableDescIfAs
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1273
*   | github.com/cockroachdb/cockroach/pkg/sql.(*createTableNode).startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:383
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:524
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:299
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
* Wraps: (6) could not find matching function overload for given arguments
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) *pgerror.withCandidateCode (5) *withstack.withStack (6) *errutil.leafError
*
panic: could not find matching function overload for given arguments [recovered]
	panic: could not find matching function overload for given arguments [recovered]
	panic: could not find matching function overload for given arguments

goroutine 5102 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc004658008, {0x836b750, 0xc004841e90}, {0x6358720?, 0xc0068942a0?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1234 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x836b750, 0xc004841e90}, {0xc00447dbe0?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:961 +0x45
panic({0x6358720?, 0xc0068942a0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:127 +0x554
panic({0x6358720?, 0xc0068942a0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetSequenceFromFunc(0xc0040ece10)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:106 +0x1fa
github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences.func1({0x836be50, 0xc0040ece10})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:162 +0x65
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre(0xc003429d58, {0x836be50, 0xc0040ece10})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119 +0x42
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr({0x8331ea0, 0xc003429d58}, {0x836be50, 0xc0040ece10})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837 +0x35
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RangeCond).Walk(0xc0027a1ef0, {0x8331ea0, 0xc003429d58})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:501 +0x6a
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr({0x8331ea0, 0xc003429d58}, {0x836c908, 0xc0027a1ef0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:840 +0x4c
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleVisit({0x836c908, 0xc0027a1ef0}, 0xc001a786f0)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2138 +0x7b
github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr.GetUsedSequences({0x836c908, 0xc0027a1ef0})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/seqexpr/sequence.go:157 +0x85
github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang.func1({0x836c908, 0xc0027a1ef0})
	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:456 +0x6b
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*simpleVisitor).VisitPre(0xc003429ce0, {0x836c908, 0xc0027a1ef0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2119 +0x42
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr({0x8331ea0, 0xc003429ce0}, {0x836c908, 0xc0027a1ef0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:837 +0x35
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrameBound({0x8331ea0?, 0xc003429ce0?}, 0xc003429c98)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:250 +0x45
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowFrame({0x8331ea0, 0xc003429ce0}, 0xc006894280)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:221 +0x45
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkWindowDef({0x8331ea0, 0xc003429ce0}, 0xc003b017a0)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:288 +0x276
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*SelectClause).walkStmt(0xc002bddea0, {0x8331ea0, 0xc003429ce0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1844 +0x53a
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt({0x8331ea0?, 0xc003429ce0?}, {0x8392a60, 0xc002bddea0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104 +0x56
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt(0xc0027cc000, {0x8331ea0, 0xc003429ce0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718 +0x65
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt({0x8331ea0?, 0xc003429ce0?}, {0x83924e0, 0xc0027cc000})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104 +0x56
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ParenSelect).walkStmt(0xc003a85330, {0x8331ea0?, 0xc003429ce0?})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1588 +0x36
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt({0x8331ea0?, 0xc003429ce0?}, {0x83926e0, 0xc003a85330})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104 +0x56
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*Select).walkStmt(0xc0027cc050, {0x8331ea0, 0xc003429ce0})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:1718 +0x65
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.walkStmt({0x8331ea0?, 0xc003429ce0?}, {0x83924e0, 0xc0027cc050})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2104 +0x56
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.SimpleStmtVisit({0x83924e0, 0xc0027cc050}, 0xc004e02570)
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:2151 +0x7b
github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDsLang({0x836b750, 0xc0054fd830}, {0x83d4278, 0xc0046586b8}, {0xc00282d200, 0x77}, 0x0, 0x1)
	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:496 +0x50d
github.com/cockroachdb/cockroach/pkg/sql.replaceSeqNamesWithIDs(...)
	github.com/cockroachdb/cockroach/pkg/sql/create_view.go:438
github.com/cockroachdb/cockroach/pkg/sql.getFinalSourceQuery({{0x836b750?, 0xc0054fd830?}, 0xc0077b7008?, 0xc0046586b8?}, 0xc0068bda40, 0xc004658bc0)
	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1183 +0x265
github.com/cockroachdb/cockroach/pkg/sql.newTableDescIfAs({{0x836b750?, 0xc0054fd830?}, 0xc0077b7008?, 0xc0046586b8?}, 0xc003481540, {0x841ead0, 0xc003ce4f20}, {0x841b670, 0xc008667080}, 0x71, ...)
	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:1273 +0x52d
github.com/cockroachdb/cockroach/pkg/sql.(*createTableNode).startExec(0xc0054fd5c0, {{0x836b750, 0xc0054fd830}, 0xc0077b7008, 0xc0046586b8})
	github.com/cockroachdb/cockroach/pkg/sql/create_table.go:383 +0x19a5
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2({0x6a66f9b?, 0x17?}, {0x836eb98, 0xc0054fd5c0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:524 +0x13d
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112 +0x38
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00447acc8, {0x836eb98?, 0xc0054fd5c0}, {0x6a1e8b8?, 0xc?})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:299 +0x1602
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc00447acc8, {0x836eb98, 0xc0054fd5c0})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf0
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
github.com/cockroachdb/cockroach/pkg/sql.startExec({{0x836b750?, 0xc0054fd830?}, 0xc0077b7008?, 0xc0046586b8?}, {0x836eb98, 0xc0054fd5c0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:527 +0x10b
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc0064d3508, {0x836b750, 0xc0054fd800})
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:171 +0x10d
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:116 +0x2e
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc003de5950?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:152 +0x57
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start(0xc003de5950, {0x836b718?, 0xc0027a1a40?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:115 +0x7b
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc003de5950, {0x836b718?, 0xc0027a1a40?}, {0x83360e0, 0xc004be1408})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:731 +0x4a
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc0027c4f08, {0x836b718, 0xc0027a1a40}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:579 +0x22c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0x83ee580?, {0x836b718?, 0xc0027a1a40?}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:305 +0x225
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc002ae1760, {0x836b750, 0xc0054fd5f0}, 0xc0025da4e0, 0xc003481900, 0xc001ca5c80, 0xc004be1408, 0xc004658bc0, 0xc00447bad0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:932 +0xc07
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc002ae1760, {0x836b750, 0xc0054fd5f0}, 0xc004658bc0, 0xc0025da4e0, 0xc003481900, {{0x836eb98, 0xc0054fd5c0}, 0x0}, 0xc004be1408, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1994 +0x215
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0x4648a5?, 0x0?, {0x836b750?, 0xc0054fd5f0?}, 0x793685547678?, 0x10?, 0x7936cdcb4878?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1708 +0xbb
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc002ae1760, {0x836b750, 0xc0054fd5f0}, 0xc004658bc0, 0xc0025da4e0, 0xc0046586b8, 0xc004be1408, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1711 +0x225
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc004658008, {0x836b750, 0xc0054fd5f0}, 0xc0046586b8, 0x1?, {0x79368400c878?, 0xc00417ef48?}, 0x0, 0xc0069d9418, {0x831ea20, ...})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2427 +0x485
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc004658008, {0x836b718, 0xc0068bda90}, 0xc0046586b8, {0x79368400c878, 0xc00417ef48})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1974 +0x115e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc004658008, {0x836b750, 0xc00126c030}, {{0x8392760, 0xc003481540}, {0x0, 0x0, 0x0}, {0xc002724e7f, 0x80}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1181 +0x46a5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x836b750?, 0xc00126c030?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:146 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xc21f180?, {0x836b750?, 0xc00126c030?}, {0x8392760?, 0xc003481540?}, 0xc00417ef48?, 0x6a2dad5?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3436 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc004658008, {0x836b750, 0xc00126c030}, {{0x8392760, 0xc003481540}, {0x0, 0x0, 0x0}, {0xc002724e7f, 0x80}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:145 +0x709
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc00447d820, 0xc00447d8d0, 0xc004658008, 0xc00447d4f8, 0xc00447d958, 0xc00447d948, 0xc00447d978, 0xc00447d990)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2316 +0x3f1
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc004658008)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2321 +0xa39
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc004658008, {0x836b750, 0xc004841e90}, 0x3c?, 0xc0019cfd00?, 0xc0016539c0?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2238 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0025848c0?, {0x836b750?, 0xc004841e90?}, {0xc002a22900?}, 0x2?, 0xc004841830?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:963 +0xda
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0xc00417e808, {0x836b750, 0xc004841e90}, {0x0, 0x4, {0x1, {0x69e7fc3, 0x3}, {0xc00017bd00, 0xf}, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:256 +0x4a6
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func4()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1136 +0xec
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 5127
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:1133 +0x5db
```

**Environment:**

- CockroachDB version: v24.2.4
- Server OS: Ubuntu 20.04 LTS
- Client app: CockroachDB demo command. (./cockroach demo)
- Detailed CockroachDB version output:

```
cockroach version details:
Build Tag:        v24.2.4
Build Time:       2024/10/14 17:20:56
Distribution:     CCL
Platform:         linux amd64 (x86_64-pc-linux-gnu)
Go Version:       go1.22.5 X:nocoverageredesign
C Compiler:       gcc 6.5.0
Build Commit ID:  8140b8e294fb16a293d0aae662bcf5c1d488c852
Build Type:       release
Enabled Assertions: false
```

Jira issue: CRDB-43584

--- a/pkg/sql/catalog/seqexpr/sequence.go
+++ b/pkg/sql/catalog/seqexpr/sequence.go
@@ -80,10 +80,10 @@ func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (*SeqIdentifier, error) {
 			if len(funcExpr.Exprs) == overload.Types.Length() {
 				paramTypes, ok := overload.Types.(tree.ParamTypes)
 				if !ok {
-					panic(pgerror.Newf(
+					return nil, pgerror.Newf(
 						pgcode.InvalidFunctionDefinition,
 						"%s has invalid argument types", funcExpr.Func.String(),
-					))
+					)
 				}
 				found = true
 				for i := 0; i < len(paramTypes); i++ {
@@ -98,10 +98,10 @@ func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (*SeqIdentifier, error) {
 			}
 		}
 		if !found {
-			panic(pgerror.New(
+			return nil, pgerror.New(
 				pgcode.DatatypeMismatch,
 				"could not find matching function overload for given arguments",
-			))
+			)
 		}
 	}
 	return nil, nil

--- a/pkg/sql/logictest/testdata/logic_test/create_table
+++ b/pkg/sql/logictest/testdata/logic_test/create_table
@@ -1097,3 +1097,16 @@ statement ok
 SET inject_retry_errors_enabled=false
 
 subtest end
+
+
+# Addresses a bug where parsing nextval expressions with extra values could end
+# end up with a panic when rewriting sequence expressions.
+subtest 133399
+
+statement ok
+CREATE TABLE v_133399 (c01 INT);
+
+statement error pgcode 42804 could not find matching function overload for given arguments
+CREATE TABLE t_133399 AS (SELECT * FROM v_133399 WINDOW window_name AS (ROWS c01 BETWEEN nextval ('abc', 'abc', 'abc') AND c01 PRECEDING));
+
+subtest end

